BlueprintUI/Sources/Element/ElementState.swift

```diff
@@ -464,23 +464,34 @@ final class ElementState {
                 /// Because we're returning a cached layout, we're not going to be
                 /// enumerating every child element in the tree during layout. To resolve
                 /// this, we'll perform an enumeration over the tree using our cached layout values.
-                /// This allows us to update the cached `element.latest`, in case the
-                /// `backingViewDescription` has changed.
                 elementContent.forEachElement(
                     in: size,
                     with: environment,
                     children: existing.nodes,
                     state: self,
-                    forEach: { state, element, node in
+                    forEach: { state, element, node, childNodes in
+
+                        /// Guarantees that the latest element is present in the tree.
+                        /// in case its `backingViewDescription` has changed,
+                        /// for example (while remaining equivalent), eg it needs to apply
+                        /// a new callback closure to the view.
                         state.element.latest = element
 
                         /// Because we won't be visiting any child elements
                         /// for a `ComparableElement` during either
                         /// measurement or layout, mark all our child nodes
                         /// as visited so they are not torn down.
                         state.wasVisited = true
+
+                        /// Because we're returning a cached layout, we need to build our
+                        /// ordered children based on the layout nodes.
+                        state.buildOrderedChildrenIfNeeded(from: childNodes)
```

Comment:
`orderedChildren` are only used for testing/debugging correct?

Reply:
Yeah, it's used for the `identifierTree`; so we can verify the state of the tree; though it is useful; so it might get used for real eventually.

Reply:
gotcha. Just wondering if there's a way to make it no-op in release for now, it adds a bit more memory overhead that isn't being used. It can build up.

Reply:
In theory it's only a pointer's worth per child (since it's an array to classes; so 1000 elements is 8kb, even 10,000 is only 80kb); so I'd rather avoid #if-ing it out for now – but I think this is something we should test and check as we do more in-depth profiling